src/lib/y2storage/boot_requirements_strategies/base.rb

```diff
@@ -117,6 +122,17 @@ def boot_partition_needed?
         false
       end
 
+      def too_small_boot?
+        # for other partitions it is not needed as packager check disk usage, but /boot is special
+        # as it contain initrd that is generated and also bootloader code.
+        filesystem = devicegraph.filesystems.find { |f| f.mount_path == "/boot" }
+        return false unless filesystem
+
+        available_size = filesystem.blk_devices.map(&:size).reduce(:+)
+        # FIXME: it does not count fs size and other and it is just hard coded.
```

Comment:
We should actually check for available space in the filesystem. Not for the size of the container device. Is not only more precise, but also more straightforward than summing up sizes.

Just imagine we are reusing a boot partition without having explicitly marked it to be formatted - the device would be big enough but potentially full with whatever garbage already.

Reply:
BTW, a filesystem spread over several blk devices is a deprecated feature in storage-ng

Reply:
Even further, `Devicegraph#filesystems` return `Array<Filesystems::Base>` and `#blk_devices` is only defined in `Filesystems::BlkFilesystem`, not in the base class.

Reply:
so I need to use detect_size_info, right?

Reply:
Apart of commented by ancor, I see that we could check `/boot` twice. Here the available size is checked, but `/boot` in also checked in cases where it is a required separate partition, for example [here](https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2storage/boot_requirements_strategies/legacy.rb#L170).

I don't think that it is a problem, but a little weird. We could show an initial error because `boot` is too small and then other warning because the `boot` has not correct filesystem type, for example.

Reply:
@joseivanlopez as this is error warning won't be seen. We first force to fix fatal errors and then user can confirm warnings.

Reply:
and if there are more requirements for /boot partition I am not sure how we can handle it differently? join our related part for it?

Reply:
@jreidinger I guess a wrong filesystem type should also be a stopper error, isn't it? In that case we should add something as `wrong_boot_fs_type?` and check that `!Filesystems::Type.root_filesystems.include?(boot.filesystem.type)`.